sdk/src/main/java/io/radar/sdk/util/RadarSimpleLogBuffer.kt

```diff
@@ -3,59 +3,193 @@ package io.radar.sdk.util
 import io.radar.sdk.Radar
 import io.radar.sdk.model.RadarLog
 import java.util.concurrent.LinkedBlockingDeque
+import android.content.Context
+import io.radar.sdk.RadarSettings
+import org.json.JSONObject
+import java.io.File
+import java.lang.Integer.min
+import java.util.Date
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
 
 /**
- * Basic Log Buffer implementation that is backed by a field.
+ * Log Buffer implementation that persist.
  */
-internal class RadarSimpleLogBuffer : RadarLogBuffer {
+internal class
+RadarSimpleLogBuffer(override val context: Context): RadarLogBuffer{
 
     private companion object {
-        const val MAXIMUM_CAPACITY = 1000
+
+        const val MAXIMUM_MEMORY_CAPACITY = 200
+
+        const val MAXIMUM_CAPACITY = 500
         const val HALF_CAPACITY = MAXIMUM_CAPACITY / 2
+        const val logFileDir = "radar_logs"
+        var fileCounter = 0
+
+    }
+    private var persistentLogFeatureFlag = false
+
+
+    private val scheduler = Executors.newScheduledThreadPool(1)
+
+    init {
+        scheduler.scheduleAtFixedRate({ persistLogs() }, 30, 30, TimeUnit.SECONDS)
```

Comment:
Hmmm, on iOS we're not running a timer for flushing -- this is an aspect that there probably should be parity on.

Reply:
I see the iOS one -- let's make this 20s like ios